src/test/unit/engine/RESTapiTest.cpp

```diff
@@ -156,7 +156,8 @@ TEST_F(RESTapiTest, helloWorld) {
   EXPECT_STREQ(vm["result"].c_str(), "Hello World!") << "Response to GET /hi request";
 }
 
-TEST_F(RESTapiTest, example) {
+
+TEST_F(DISABLED_RESTapiTest, example) { //FIXME started segfaulting in CI Debug
```

Comment:
@dkeeney @Zbysekz do you have an idea why this would start failing, please? 
The Debug CI simply segfaults (while?) running this test: 
https://github.com/htm-community/htm.core/pull/783/checks?check_run_id=1080429326#step:5:525 

I made no changes that should directly affect any functionality of NetworkAPI or RESTAPI, but something in SP might change. 
I'm now testing w/o this test if it's the only thing. 

OK, locally I reproduce the same error: 
```
[       OK ] RESTapiTest.helloWorld (505 ms)
[ RUN      ] RESTapiTest.example
Traceback (most recent call last):
File "../../setup.py", line 448, in <module>
"htm-bindings-check = htm.bindings.check:checkMain",
File "/home/mmm/.conda/envs/htmcore/lib/python3.7/site-packages/setuptools/__init__.py", line 161, in setup
return distutils.core.setup(**attrs)
File "/home/mmm/.conda/envs/htmcore/lib/python3.7/distutils/core.py", line 148, in setup
dist.run_commands()
File "/home/mmm/.conda/envs/htmcore/lib/python3.7/distutils/dist.py", line 966, in run_commands
self.run_command(cmd)
File "/home/mmm/.conda/envs/htmcore/lib/python3.7/distutils/dist.py", line 985, in run_command
cmd_obj.run()
File "/home/mmm/.conda/envs/htmcore/lib/python3.7/site-packages/setuptools/command/test.py", line 238, in run
self.run_tests()
File "../../setup.py", line 172, in run_tests
subprocess.check_call([cpp_tests])
File "/home/mmm/.conda/envs/htmcore/lib/python3.7/subprocess.py", line 328, in check_call
raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/mnt/store/devel/HTM/htm-community/nupic.cpp/build/Release/bin/unit_tests']' died with <Signals.SIGSEGV: 11>.
```

Does not say much :(, is there anything in bindings that should be altered too?

Reply:
I can confirm that it crashes in debug only, release is OK.

It crashes right [here](https://github.com/htm-community/htm.core/blob/0c275834ba983fb304dd7effc6b9d14b7e51399d/src/test/unit/engine/RESTapiTest.cpp#L217)
It is because `client->Get(message) `returned `null_ptr`. What i briefly saw inside http lib, `null_ptr `is possible in some case to return.

Reply:
thank you for looking into it! :+1: 

> I can confirm that it crashes in debug only, release is OK.

strange it manifests only now. do you think it's something with httplib compiled in debug? (same as we don't do Py bindings in deb.), ie. should we just skip the test in Debug?

Reply:
hmm i am trying to understand this simple example from the lib
(https://github.com/yhirose/cpp-httplib/blob/master/example/simplecli.cc)[https://github.com/yhirose/cpp-httplib/blob/master/example/simplecli.cc] i don't know res.error() could work. res is shared ptr. It throws 
`error: ‘class std::shared_ptr<httplib::Response>’ has no member named ‘error’` on me

Reply:
to get some error description, why the Get() returns null

Reply:
> strange it manifests only now. do you think it's something with httplib compiled in debug? (same as we don't do Py bindings in deb.), ie. should we just skip the test in Debug?

But then why it doesn't fail in master at the start of this branch right? CI was not changed right?

Reply:
> But then why it doesn't fail in master at the start of this branch right? CI was not changed right? 

CI wasn't changed. I just checked the diff to master, this PR only touches internals of SP local inhibition (`SP.inhibitColumnsLocal_()`, `Topology.hpp`). Not even SP's "public API" was changed. 

> Maybe it just revealed some leakage in this library... from github it seems that they are making bugfixes. I will try newest version of that lib 

cool, that'll be nice. If you'd just make a new PR, as it should pass master, and I'll then get it merged here.

Reply:
The bug in RESTapi must be something unrelated.  It should not be too hard to find.

Reply:
I'd suggest we see if the update to cpp-httplib (there seems to be a lot of news) magically fixes our problem. If not, I'd just enable the test in Release and we can keep a separate issue for solving the Debug segfault later. If that sounds ok?

Reply:
ok yes. I already compile it,
and update to latest didn't solve it..

Reply:
Followup issue in httplib please in https://github.com/htm-community/htm.core/issues/884